File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def check_scoville(arr, K):
    for item in arr:
        if item < K:
            return False
    return True # 모두 K 이상

def solution(scoville, K):
    heapq.heapify(scoville)
    count = 0
    
    while(len(scoville) > 1):
        if check_scoville(scoville, K):
            return count
        
        first = heapq.heappop(scoville)
        second = heapq.heappop(scoville)
        
        value = first + second * 2
        
        heapq.heappush(scoville, value)
        
        count += 1
    
    if check_scoville(scoville, K):
        return count
    
    return -1

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> score = new PriorityQueue<>();
        for (int i : scoville) {
            score.offer(i);
        }
        
        int cnt = 0;
        
        while (score.peek() < K) {
            if (score.size() < 2) {
                return -1;
            }
            
            int first = score.poll();
            int second = score.poll();
            score.offer(first + second * 2);
            cnt++;
        }
        return cnt;
    }
}